extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -324,29 +370,56 @@ function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
       madeInClient = false
     ) => {
       const enableStudyLazyLoad = true;
-      wadoDicomWebClient.headers = generateWadoHeader();
+      const clients = clientManager.getClientsForWadoRequests();
       // Get Series
-      const { preLoadData: seriesSummaryMetadata, promises: seriesPromises } =
-        await retrieveStudyMetadata(
-          wadoDicomWebClient,
-          StudyInstanceUID,
-          enableStudyLazyLoad,
-          filters,
-          sortCriteria,
-          sortFunction
-        );
+      let seriesSummaryMetadata = [];
+      let seriesPromises = [];
+      const seriesClientsMapping = {};
+      for (let i = 0; i < clients.length; i++) {
+        let clientSeriesSummaryMetadata, clientSeriesPromises;
+        try {
+          const { preLoadData, promises } = await retrieveStudyMetadata(
+            clients[i].wadoDicomWebClient,
+            StudyInstanceUID,
+            enableStudyLazyLoad,
+            filters,
+            sortCriteria,
+            sortFunction
+          );
+          clientSeriesSummaryMetadata = preLoadData;
+          clientSeriesPromises = promises;
+        } catch {
+          clientSeriesSummaryMetadata = [];
+          clientSeriesPromises = [];
+        }
+
+        // create a mapping between SeriesInstanceUID <--> clientName, for two reasons:
+        // 1 - remove duplicates in series metadata
+        // 2 - associate each instance in a series with the name of the client it was retrieved
+        for (let j = 0; j < clientSeriesSummaryMetadata.length; j++) {
+          if (!seriesClientsMapping[clientSeriesSummaryMetadata[j].SeriesInstanceUID]) {
+            seriesClientsMapping[clientSeriesSummaryMetadata[j].SeriesInstanceUID] =
+              clients[i].name;
+
+            seriesSummaryMetadata.push(clientSeriesSummaryMetadata[j]);
+            seriesPromises.push(clientSeriesPromises[j]);
+          }
+        }
```

Comment:
for (const [j, seriesSummary] of clientSeriesSummaryMetadata.entries()) {
  const seriesUID = seriesSummary.SeriesInstanceUID;

  if (!seriesClientsMapping[seriesUID]) {
    seriesClientsMapping[seriesUID] = clients[i].name;
    seriesSummaryMetadata.push(seriesSummary);
    seriesPromises.push(clientSeriesPromises[j]);
  }
}